llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -10572,18 +10572,23 @@ bool X86InstrInfo::getMachineCombinerPatterns(
   case X86::VPDPWSSDrm:
   case X86::VPDPWSSDYrr:
   case X86::VPDPWSSDYrm: {
-    Patterns.push_back(MachineCombinerPattern::DPWSSD);
-    return true;
+    if (!Subtarget.hasFastPWSSD()) {
```

Comment:
Put this at the beginning?
```
if (!Subtarget.hasFastPWSSD())
  return TargetInstrInfo::getMachineCombinerPatterns(Root, Patterns,
                                                     DoRegPressureReduce);
```

Reply:
Can you remove the default, break if we don't use the MachineCombinerPattern::DPWSSD pattern and move this to after the switch:
```
return TargetInstrInfo::getMachineCombinerPatterns(Root, Patterns, DoRegPressureReduce);
```